### PR TITLE
feat: initial CLI layout

### LINKED ISSUE #1: cli: create initial stub
Create basic CLI app, using only the std library, to take in:

1) a URL to a project to analyze
2) path to a locally checked-out project

In both cases, the project will search for all the `go.mod` in the project

Have the flags:

* `--json` to output the final report in json
* `--debug` to output additional verbose logs for debugging

Examples:

```
gograveyard run https://github.com/influxdata/telegraf
gograveyard run telegraf
```

Open questions:

1) Should this be behind a "verb" like run or just execute as part of the root subcommand? For example, what if we added at a later date additional subcommands? (e.g. `gograveyard run <file>`)
2) Should the URL and the local path be direct to the go.mod file itself? Or should we assume it is in the root of that directory? For example, what if a project has a go.mod or subprojects?

### PR DESCRIPTION
**Pull request by powersj**
Adds the ability to take on flags and subcommands for various options.
Consider this a template, an initial layout to get us started.

This is based on the code in the #9.

fixes: #1

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    # specify parallel on every test
+    - paralleltest
+    # tests in a separate package
+    - testpackage
+    # short variable names
+    - varnamelen
+    # whitespace linter
+    - wsl
+linters-settings:
+  lll:
```
Comment (sspaink) [inline .golangci.yml]: What does `lll` stand for?

Comment (powersj) [reply .golangci.yml]: [line length linter](https://golangci-lint.run/usage/linters/#lll) I think

Comment (powersj) [reply .golangci.yml]: if you want to use the default 120 that works too

Comment (sspaink) [reply .golangci.yml]: Thanks for the doc link, I couldn't find it and thought maybe it was a typo lol

100 or 120 are both fine by me.

**Review by sspaink — APPROVED** [review]

**Pull request merged**